src/headless/converse-muc.js

```diff
@@ -917,7 +917,17 @@ converse.plugins.add('converse-muc', {
                 const msgid = message.getAttribute('id'),
                       jid = message.getAttribute('from');
                 if (msgid) {
-                    return this.messages.where({'msgid': msgid, 'from': jid}).length;
+                    const length = this.messages.where({'msgid': msgid, 'from': jid}).length;
+                    if (length > 0) {
+                        message = this.messages.findWhere({msgid});
+                        if (!message || message.get('received')) {
+                            return length;
+                        }
+                        message.save({
+                            'received': moment().format()
+                        });
+                    }
+                    return length;
```

Comment:
Can you please explain the point of this block of code?

It's not clear to me why this is necessary when you already added a `handleReceipt` method which you call in `onMessage`. You might as well call that in `onMessage` in `converse-muc.js` as well.

What also confuses me is that AFAICT receipts won't show up as duplicates because the "duplicate" msgid is on the `receipt` element, and not the `message` element.

Reply:
The idea is not to use receipts at all but simply indicate the reception of the message reflection from the muc as a receipt to the user. This mimics the behaviour of Conversations btw.

Reply:
Ok, I understand now thanks.